### ISSUE TITLE
Add support for publisher purposes consent string

### ISSUE THREAD

**Issue by nhagen** [InteractiveAdvertisingBureau/Consent-String-SDK-JS#18]
Reading through the documentation, it doesn't appear that this library has any support for the Publisher Consents component of the consent string spec. Has there been any discussion on this?

Related, from the Java SDK: https://github.com/InteractiveAdvertisingBureau/Consent-String-SDK-Java/issues/12

>... There are two distinct consent string formats outlined in the specification (v.1.1):
>1. global, vendor consent string (euconsent) as described in https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/blob/master/Consent%20string%20and%20vendor%20list%20formats%20v1.1%20Final.md#vendor-consent-string-format-
>2. publisher purposes consent string (pubconsent) as described in https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/blob/master/Consent%20string%20and%20vendor%20list%20formats%20v1.1%20Final.md#publisher-purposes-consent-string-format-
>
>However, currently the library appears to support only global, vendor consent string (correct me if I'm wrong). Looking from a publisher's perspective it would surely help if the publisher purposes consent string was supported as well. It makes sense since both consent strings have similar formats so there is potential for reusing significant part of codebase between the two implementations.
> ...

**Comment by chrispaterson** [comment]
You're right it doesn't.  We should support this, but no one has offered up code yet.  If you want to take a stab at it and put in a pull request we could do that.

**Comment by chrispaterson** [comment]
@nhagen are you still interested in this?

**Comment by chrispaterson** [comment]
Sorry didn't mean to close, clicked the wrong button.  The Internet is hard...

**Comment by amarocolas** [comment]
Hey guys, I have started this, can already get the data from the cookie

https://github.com/InteractiveAdvertisingBureau/Consent-String-SDK-JS/pull/30
